catalog/PartitionSchemeHeader.hpp

```diff
@@ -288,22 +310,46 @@ class RangePartitionSchemeHeader : public PartitionSchemeHeader {
    * @brief Check if the partition range boundaries are in ascending order.
    **/
   void checkPartitionRangeBoundaries() {
+    for (const PartitionValues &partition_range_boundary : partition_range_boundaries_) {
+      CHECK_EQ(partition_attribute_ids_.size(), partition_range_boundary.size())
+          << "A partition boundary has different size than that of partition attributes.";
+    }
+
     for (partition_id part_id = 1; part_id < partition_range_boundaries_.size(); ++part_id) {
-      if (less_unchecked_comparator_->compareTypedValues(
-              partition_range_boundaries_[part_id],
-              partition_range_boundaries_[part_id - 1])) {
-        LOG(FATAL) << "Partition boundaries are not in ascending order.";
+      CHECK(lessThan(partition_range_boundaries_[part_id - 1], partition_range_boundaries_[part_id]))
+          << "Partition boundaries are not in ascending order.";
+    }
+  }
+
+  bool lessThan(const PartitionValues &lhs, const PartitionValues &rhs) const {
+    DCHECK_EQ(partition_attribute_ids_.size(), lhs.size());
+    DCHECK_EQ(partition_attribute_ids_.size(), rhs.size());
+
+    for (std::size_t attr_index = 0; attr_index < partition_attribute_ids_.size(); ++attr_index) {
+      if (less_unchecked_comparators_[attr_index]->compareTypedValues(lhs[attr_index], rhs[attr_index])) {
+        break;
```

Comment:
Do you mean to use ``continue`` here? The break will cause the control to come out of the for loop and we wouldn't have compared all the attributes, subsequently returning true. Am I missing something?